samza-rest/src/main/java/org/apache/samza/monitor/JobsClient.java

```diff
@@ -90,41 +91,45 @@ public JobStatus getJobStatus(JobInstance jobInstance) {
    * When a job status server is down or returns a error response, it tries to reach out to
    * the next job status server in the sequence, to complete the http get request.
    *
-   * @param urlMapFunction to build the request url, given job status server base url.
+   * @param requestUrlBuilder to build the request url, given job status server base url.
    * @param <T> return type of the http get response.
    * @return the response from any one of the job status server.
    * @throws Exception when all the job status servers are unavailable.
    *
    */
-  private <T> T retriableHttpGet(Function<String, String> urlMapFunction) {
+  private <T> T retriableHttpGet(Function<String, String> requestUrlBuilder, TypeReference<T> typeReference) {
     Exception fetchException = null;
     for (String jobStatusServer : jobStatusServers) {
-      String requestUrl = urlMapFunction.apply(jobStatusServer);
+      String requestUrl = requestUrlBuilder.apply(jobStatusServer);
       try {
         ObjectMapper objectMapper = new ObjectMapper();
-        return objectMapper.readValue(httpGet(requestUrl), new TypeReference<T>() {});
+        return objectMapper.readValue(httpGet(requestUrl), typeReference);
       } catch (Exception e) {
         LOG.error(String.format("Exception when fetching tasks from the url : %s", requestUrl), e);
         fetchException = e;
       }
     }
-    throw new SamzaException(String.format("Exception during http get from urls : %s", jobStatusServers),
-        fetchException);
+    throw new SamzaException(String.format("Exception during http get from urls : %s", jobStatusServers), fetchException);
```

Comment:
Message is misleading. Implies the same exception from each server. Either keep and log a Map<url, exception> or make message clearer.